orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -2115,7 +2115,59 @@ Example 4:
         // The bias tensor has the same shape of the scale tensor.
         propagateElemTypeFromInputToOutput(ctx, 2, 2);
         propagateShapeFromInputToOutput(ctx, 2, 2);
-      });
+      })
+      .SetContextDependentFunctionBodyBuilder(
+          [](const FunctionBodyBuildContext& ctx, const OpSchema& schema, FunctionProto& functionProto) {
+            FunctionBuilder builder(functionProto);
+
+            auto* tp = ctx.getInputType(0);
+            if ((tp == nullptr) || (!tp->has_tensor_type()))
+              return false;
+            int64_t T = tp->tensor_type().elem_type();
+
+            // Requirements/assumptions:
+            // Inputs Y_grad and X are of shape [d[0], ..., d[axis-1], d[axis], ..., d[rank-1]] and type T
+            // Input scale is of shape [d[axis], ..., d[rank-1]] and type U
+            // Inputs mean and inv_std_dev are of shape [d[0], ..., d[axis-1], 1, ..., 1] (same rank as X)
+            // and type U.
+            //
+            auto axis_ref_attr = MakeRefAttribute("axis", AttributeProto_AttributeType::AttributeProto_AttributeType_INT);
+            builder
+                .AddOpset("", 15)
+                .Add("cast_mean = Cast (mean)", "to", T)
+                .Add("cast_inv_std_dev = Cast(inv_std_dev)", "to", T)
+                .Add("x_2d = Flatten (X)", axis_ref_attr)
+                .Add("Y_grad_2d = Flatten (Y_grad)", axis_ref_attr)
+                .Add("mean_2d = Flatten (cast_mean)", axis_ref_attr)
+                .Add("inv_std_dev_2d = Flatten (cast_inv_std_dev)", axis_ref_attr)
+                .Add(R"ONNX(
+                  shape_x = Shape (X)
+                  bias_scale_shape = Shape (scale)
+                  scale_2d = Flatten <axis = 0> (scale)
+
```

Comment:
this is cool, so we can now parse those expression to onnx proto?

Reply:
yes